hyperspy/io_plugins/fei.py

```diff
@@ -503,17 +532,22 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
                             'Dim-%i_CalibrationOffset' % (i + 1)][0],
                         'scale': header[
                             'Dim-%i_CalibrationDelta' % (i + 1)][0],
-                        'units': header['Dim-%i_Units' % (i + 1)][0],
+                        # for image stack, the UnitsLength is 0 (no units)
+                        'units': header['Dim-%i_Units' % (i + 1)][0] if header['Dim-%i_UnitsLength'%(i+1)]>0 else 'a.u.',
                         'size': header['Dim-%i_DimensionSize' % (i + 1)][0],
                     })
                 array_shape.append(header['Dim-%i_DimensionSize' % (i + 1)][0])
+        units = "Unknown"
+        if objects is not None:
+            objects_dict = convert_xml_to_dict(objects[0])
+            units = guess_units_from_mode(objects_dict, header)
```

Comment:
Ok, this is right but it actually seems to work!
Even if I don't exactly know why, I guess this is mainly because the information retrieve from the xml and that is used by  "guess_units_from_mode" is only the mode, which is present in almost all xml chunks, except for spectrum image, where there are more xml chunks than ser files and the additional ones doesn't contains any information.
Since, this function is useful to get the unit particularly in case of diffraction, I will find a workaround to consider the case where the xml chunk is empty and then return "Unknow" as unit.